examples/cifar10/pt/learners/cifar10_learner.py

```diff
@@ -155,17 +156,19 @@ def _initialize_trainer(self, fl_ctx: FLContext):
 
         self.valid_loader = torch.utils.data.DataLoader(self.valid_dataset, batch_size=64, shuffle=False, num_workers=2)
 
+        self.initialized = True
+
     def _terminate_trainer(self):
         # collect threads, close files here
         pass
 
     def handle_event(self, event_type: str, fl_ctx: FLContext):
         # the start and end of a run - only happen once
-        if event_type == EventType.START_RUN:
+        if event_type == EventType.START_RUN:  # TODO: Is this still needed? Clara's FederatedLearner just used init in train/val functions
             try:
-                self._initialize_trainer(fl_ctx)
+                self._initialize(fl_ctx)
             except BaseException as e:
-                error_msg = f"Exception in _initialize_trainer: {e}"
+                error_msg = f"Exception in _initialize: {e}"
                 self.log_exception(fl_ctx, error_msg)
                 self.system_panic(error_msg, fl_ctx)
         elif event_type == EventType.END_RUN:
```

Comment:
Similarly, you shouldn't handle the END_RUN event. Instead you should implement the finalize() method.

Reply:
Ok. Used finalize() method now.